src/ClientHandle.cpp

```diff
@@ -318,84 +322,100 @@ void cClientHandle::Kick(const AString & a_Reason)
 
 void cClientHandle::Authenticate(const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties)
 {
-	if (m_State != csAuthenticating)
+	cWorld * World;
 	{
```

Comment:
It's simple RAII - acquire resource in constructor, release in destructor. C++ guarantees the destructor to be called upon scope exit.